pysensors/basis/_custom.py

```diff
@@ -0,0 +1,82 @@
+"""
+custom mode basis class.
+"""
+from ._base import InvertibleBasis
+from ._base import MatrixMixin
+
+class Custom(InvertibleBasis, MatrixMixin):
+    """
+    Generate a custom transformation which maps input features to
+    custom modes.
+
+    Assumes the data has already been centered (to have mean 0).
+
+    Parameters
+    ----------
+    n_basis_modes : int, optional (default 10)
+        Number of basis modes to retain. Cannot be larger than
+        the number of features ``n_features``, or the number of examples
+        ``n_examples``.
+
+    Attributes
+    ----------
+    basis_matrix_ : numpy ndarray, shape (n_features, n_basis_modes)
+        The top n_basis_modes left singular vectors of the training data.
+
+    """
+
+    def __init__(self, n_basis_modes=10,U = None, **kwargs):
+        if isinstance(n_basis_modes, int) and n_basis_modes > 0:
+            super(Custom, self).__init__()#n_components=n_basis_modes, **kwargs
+            self._n_basis_modes = n_basis_modes
+            self.custom_basis_ = U
+        else:
+            raise ValueError("n_basis_modes must be a positive integer.")
+
+    def fit(self, X):
+        """
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The training data.
+
+        Returns
+        -------
+        self : instance
+        """
+        # self.basis_matrix_ = super(Custom, self).fit(X).components_.T
+        self.basis_matrix_ = self.custom_basis_
```

Comment:
Possibly we should check that n_samples and n_features from X are greater than n_basis_nodes.

Reply:
Also, maybe we should just set the basis_matrix_ in the ```__init__``` function.

Reply:
I thought about that but That's not what they did in the other basis options. Of course, because the other bases are computed from X. I will try and see what happens.

Reply:
Okay.